pkg/verification/service/verification_service_test.go

```diff
@@ -159,6 +186,40 @@ func (s *TestVerificationServiceSuite) TestInitVerification() {
 		params.Get("Body"))
 	require.Equal(s.T(), "CodeReady", params.Get("From"))
 	require.Equal(s.T(), "+1NUMBER", params.Get("To"))
+
+	// Test the init verification for the second UserSignup - Setup gock again for another request
+	gock.New("https://api.twilio.com").
+		Reply(http.StatusNoContent).
+		BodyString("")
+
+	obs = func(request *http.Request, mock gock.Mock) {
+		reqBody = request.Body
+	}
+	gock.Observe(obs)
+
+	ctx, _ = gin.CreateTestContext(httptest.NewRecorder())
+	// This time we won't pass in the UserID, just the username yet still expect the UserSignup to be found
+	err = s.Application.VerificationService().InitVerification(ctx, "", userSignup2.Spec.Username, "+61NUMBER")
+	require.NoError(s.T(), err)
+
+	userSignup2, err = s.FakeUserSignupClient.Get(userSignup2.Name)
+	require.NoError(s.T(), err)
+
+	// Ensure the verification code is set
+	require.NotEmpty(s.T(), userSignup2.Annotations[toolchainv1alpha1.UserSignupVerificationCodeAnnotationKey])
+
+	buf = new(bytes.Buffer)
+	_, err = buf.ReadFrom(reqBody)
```

Comment:
don't you need to call `reqBody.Close()` in a deferred func?

Reply:
Not sure if it matters in a unit test, but I've added deferred `Close()` calls anyway.